src/Database/Schema/MysqlSchemaDialect.php

```diff
@@ -802,7 +803,9 @@ public function columnDefinitionSql(array $column): string
             unset($column['default']);
         }
         if (isset($column['comment']) && $column['comment'] !== '') {
-            $out .= ' COMMENT ' . $this->_driver->schemaValue($column['comment']);
+            // Always quote comments as strings to prevent SQL syntax errors with numeric comments
+            // See: https://github.com/cakephp/migrations/issues/889
+            $out .= ' COMMENT ' . $this->_driver->getPdo()->quote((string)$column['comment'], PDO::PARAM_STR);
```

Comment:
```suggestion
            $out .= ' COMMENT ' . $this->_driver->quote((string)$column['comment']);
```

`Driver::quote()` was added in 5.2